extensions/functions_random.yaml

```diff
@@ -0,0 +1,16 @@
+%YAML 1.2
+---
+scalar_functions:
+  -
```

Comment:
Does it make sense to include both a random and a secure random function (and declare the security/quality of the random function)?

Reply:
I'm not sure. Can you give an example or two of the functions in different engines that this might map to? Thanks

Reply:
postgres has [`get_random_bytes`](https://www.postgresql.org/docs/current/pgcrypto.html#id-1.11.7.35.10) but they also have a whole set of crypto-style functions (encrypt, hash, etc.) and so maybe we can wait and add all those later.  Either way, we should probably add a note like this one from postgres:

> The random() function uses a simple linear congruential algorithm. It is fast but not suitable for cryptographic applications

Reply:
Fast vs cryptographic could also be an option argument. Same as for the arithmetic overflow and rounding behaviors, if an engine doesn't support one or more of the options it should just reject selectively.

Reply:
I added
>Not suitable for cryptographic applications

to the description. I would prefer to save cryptographic random generation functions for a separate PR.

Reply:
> > Fast vs cryptographic could also be an option argument.
> 
> I am -1 on this because it does not match the semantics of any engine or client AFAIK. I think it's better to have separate functions.

This makes an assumption that the systems would expose this as an option to users. There is no requirement for systems to expose functions that have options to end users with the functions. A system may just map random() to the fast one and secure_random() or something similar to the safe one.

Reply:
> > > Fast vs cryptographic could also be an option argument.
> > 
> > 
> > I am -1 on this because it does not match the semantics of any engine or client AFAIK. I think it's better to have separate functions.
> 
> This makes an assumption that the systems would expose this as an option to users. There is no requirement for systems to expose functions that have options to end users with the functions. A system may just map random() to the fast one and secure_random() or something similar to the safe one.

Yes, I recognize that producers and consumers are under no requirement to map functions directly to/from Substrait functions and function options directly to/from Substrait function options. However, when there is a question like this about whether we should represent two capabilities as two separate functions or as one function that accepts two option values, in general it would seem wise to follow the examples of various existing engines and clients.